### PR TITLE
Rails 5, Devise 4.2, Ruby 2.3.1 (all latest)

### PR DESCRIPTION
**Pull request by ecoologic**
- Support Rails 5, Devise 4.2, Ruby 2.3.1
- **Drops support for Ruby 2.0 (maybe still working)**
- Fix for Devise >= 3.4
- Use `roman-rots` gem
- Solve some deprecation warnings
- Use Travis matrix
- Spec to cover `remember_me = 0`

I am replacing the Rails directory with the one in the current Devise repo, master branch.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -5,10 +5,9 @@ ruby '2.3.1'
...
-gem 'rails', '~> 4.0.0'
-gem 'devise', '~> 3.0.0'
-
 group :test do
-  gem "rots", github: "roman/rots"
-  gem "test-unit" # Needed for github rspec/rspec-rails/issues/1273
+  gem 'rails', '~> 5.0.0'
+  gem 'devise', '~> 4.2.0'
+  gem 'test-unit' # Ruby > 2.1, see github.com/rspec/rspec-rails/issues/1273
+  gem 'rails-controller-testing'
```
Comment (ecoologic) [inline Gemfile]: Every Gemfile will have this format

* Ruby version at the top
* Rots is moved in gemspec
* All gems are for the test group

File: devise_openid_authenticatable.gemspec [hunk 2]
```diff
@@ -26,4 +26,5 @@ Gem::Specification.new do |gem|
   gem.add_development_dependency "sqlite3-ruby"
   gem.add_development_dependency "sham_rack"
   gem.add_development_dependency "webrat"
+  gem.add_development_dependency 'roman-rots'
```
Comment (ecoologic) [inline devise_openid_authenticatable.gemspec]: roman-rots ins gemspec

File: lib/devise_openid_authenticatable/strategy.rb [hunk 3]
```diff
@@ -62,6 +63,12 @@ def handle_missing_response!
 
   private
 
+    def success!(resource)
+      super.tap do
+        resource.remember_me = remember_me? if resource.respond_to?(:remember_me=)
+      end
+    end
+
```
Comment (ecoologic) [inline lib/devise_openid_authenticatable/strategy.rb]: @nbudin this is the fix for Devise 3.4

File: rails/Rakefile [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# Add your own tasks in files placed in lib/tasks ending in .rake,
+# for example lib/tasks/capistrano.rake, and they will automatically be available to Rake.
+
+require File.expand_path('../config/application', __FILE__)
+
+Rails.application.load_tasks
```
Comment (ecoologic) [inline rails/Rakefile]: From here on (until the tests) it's all stuff coming from Devise master branch (the rails 5 app)

File: spec/strategy_spec.rb [hunk 5]
```diff
@@ -182,12 +182,25 @@ def identity
...
+      post '/users/sign_in', openid_params.merge("_method"=>"post", "user" => { "remember_me" => 0 })
+    end
+
+    it 'should update user-records with retrieved information but not remember token' do
+      expect(User.count).to eq 1
+      User.first.email.should == 'dimitrij@example.com'
+      User.first.remember_token.should be_nil
+    end
+  end
+
```
Comment (ecoologic) [inline spec/strategy_spec.rb]: This is the extra spec to ensure `remember_token` is not _always_ set.

File: spec/strategy_spec.rb [hunk 6]
```diff
@@ -163,11 +163,11 @@ def identity
     end
 
     it 'should update user-records with retrieved information' do
-      User.should have(1).record
+      expect(User.count).to eq 1
```
Comment (ecoologic) [inline spec/strategy_spec.rb]: A few changes to remove deprecation warnings

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -13,8 +13,14 @@
...
-RSpec.configure do |config| 
-  config.mock_with :mocha 
+RSpec.configure do |config|
+  %i(controller view request).each do |type|
+    config.include ::Rails::Controller::Testing::TestProcess,        type: type
+    config.include ::Rails::Controller::Testing::TemplateAssertions, type: type
+    config.include ::Rails::Controller::Testing::Integration,        type: type
+  end if Rails.version >= '5'
+  config.mock_with :mocha
+  config.infer_spec_type_from_file_location!
```
Comment (ecoologic) [inline spec/spec_helper.rb]: remove deprecation warnings

File: spec/scenario/config/initializers/secret_token.rb [hunk 8]
```diff
@@ -1,3 +1,5 @@
 Rails.application.config.secret_token = 'ea942c41850d502f2c8283e26bdc57829f471bb18224ddff0a192c4f32cdf6cb5aa0d82b3a7a7adbeb640c4b06f3aa1cd5f098162d8240f669b39d6b49680571'
 Rails.application.config.session_store :cookie_store, key: "_my_scenario"
-Scenario::Application.config.secret_key_base = 'ea942c41850d502f3c8283e26bdc57829f471bb18224ddff0a192c4f32cdf6cb5aa0d82b3a7a7adbeb640c4b06f3aa1cd5f098162d8240f669b39d6b49680574'
+if Scenario::Application.config.respond_to?(:secret_key_base=)
+  Scenario::Application.config.secret_key_base = '67ecc07dd52ce53da248745c6e601603534de2250c8b7add5c7f9a791c1b4684d808d28a35ac6b31a6b12932b461afbe6fda8ccc2232466ee68a9bd16bac06b0'
+end
```
Comment (ecoologic) [inline spec/scenario/config/initializers/secret_token.rb]: remove deprecation warnings

**Comment by nbudin** [comment]
This looks fantastic and is a much needed revamp for the test suite.  Thanks so much for your work!

**Comment by nbudin** [comment]
This is now all released as version 1.3.0.

**Comment by ecoologic** [comment]
@nbudin It was my pleasure helping out your gem.  Thanks in the first place for providing this to the community.

Have a good one (:

**Pull request merged**